testah-junit/src/main/java/org/testah/framework/testPlan/TestSystem.java

```diff
@@ -0,0 +1,620 @@
+
+            for (final Method m : desc.getTestClass().getDeclaredMethods()) {
+                if (null != m.getAnnotation(Ignore.class) && null != m.getAnnotation(Disabled.class)) {
+                    addIgnoredTest(desc.getDisplayName(), "JUNIT_IGNORE");
+                }
+            }
+            MDC.put("logFileName", "" + Thread.currentThread().getId());
+        }
+        TS.log().info(Cli.BAR_LONG);
+
+        TS.log().info(
+            "TESTCASE started:" + desc.getDisplayName() + " - thread[" + Thread.currentThread().getId() + "]");
+        startTestCase(desc, testCase, testPlan,
+            desc.getAnnotation(KnownProblem.class));
+        getTestStep();
+    }
+
+    /**
+     * Failed called if a testCase Fails.
+     *
+     * @param e           exception given by junit.
+     * @param description testCase description sent by junit.
+     */
+    public void failed(final Throwable e, final Description description) {
+        setDescription(description);
+        TS.step().action().createAssertResult("Unexpected Error occurred", false, "UnhandledExceptionFoundByJUnit", "",
+            e.getMessage(), e).log();
+
+        TS.log().error("TESTCASE Status: failed", e);
+        stopTestCase(false);
+    }
+
+    /**
+     * Succeeded called if testCase succeeded.
+     *
+     * @param description testCase description sent by junit.
+     */
+    public void succeeded(final Description description) {
+        setDescription(description);
+        stopTestCase((TS.params().isResultIgnoredIfNoAssertsFound() ? null : true));
+        TS.log().info("TESTCASE Status: " + getTestCase().getStatusEnum());
+        try {
+            final Test testAnnotation = description.getAnnotation(Test.class);
+            if (null != testAnnotation && Test.None.class == testAnnotation.expected()) {
+                if (null != getTestCase()) {
+                    getTestCase().getAssertionError();
+                    if (null == getTestCase().getStatus()) {
+                        addIgnoredTest(description.getDisplayName(),
+                            "NA_STATUS_NO_ASSERTS");
+                        return;
+                    }
+                }
+            }
+        } catch (final AssertionError ae) {
+            TS.log().error("Exception Thrown Looking at TestCase Assert History\n" + ae.getMessage());
+            throw ae;
+        }
+    }
+
+    /**
+     * Finished called when the testCase completes.
+     *
+     * @param desc testCase description sent by junit.
+     */
+    public void finished(final Description desc) {
+        setDescription(desc);
+        TS.log().info("TESTCASE Complete: " + desc.getDisplayName() + " - thread[" + Thread.currentThread().getId() + "]");
+        if (null == getTestCase().getStatus()) {
+            return;
+        } else if (getTestCase().getStatus()) {
+            // doOnPass(); TODO find solution
+        } else {
+            // doOnFail(); TODO find solution
+        }
```

Comment:
right now the conditionals don't do much, is there a ticket for the TODOs, are conditions necessary?

Reply:
I added in support, junit 5 implementation works different, but added a way to be able to define closures for what to do on pass or fail from the test class like in the junit 4 implementation.